manager/network/model.go

```diff
@@ -0,0 +1,15 @@
+package network
+
+import (
+	"github.com/docker/swarmkit/api"
+	"github.com/docker/swarmkit/manager/allocator/networkallocator"
+)
+
+// Model is an abstraction over the Network Model to be used.
+type Model interface {
+	Allocator() (networkallocator.NetworkAllocator, error)
+	SetDefaults(spec *api.NetworkSpec) error
+	ValidateNetworkSpec(spec *api.NetworkSpec) error
+	PredefinedNetworks() []networkallocator.PredefinedNetworkData
+	SupportIngressNetwork() bool
```

Comment:
@aaronlehmann I made the name change as an interim measure. but WRT the `IngressNetwork` method I think it might make sense to move the `newIngressNetwork` function into the cnm version of that and use the type check WDYT?

I'm also eyeing up `newPredefinedNetwork` in a similar light, in essence making `PredefinedNetworks` return an `api.Network` instead of the `PredefinedNetworkData` and so embedding the function into CNM where it is used.